packages/cli/src/commands/deploy/index.ts

```diff
@@ -848,12 +853,30 @@ const printDeploymentStatus = async (
       action?: string;
     };
   },
-  deployStamp: () => string
+  deployStamp: () => string,
+  noWait: boolean
 ) => {
   indications = indications || [];
   const isProdDeployment = target === 'production';
 
-  if (readyState !== 'READY') {
+  let stillBuilding = false;
+  if (noWait) {
+    if (
+      readyState !== 'READY' &&
+      readyState !== 'CANCELED' &&
+      readyState !== 'ERROR'
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (non-blocking):** ‏group states into named collections

We use this group of terminal states here and a different group of in-progress states [below](https://github.com/vercel/vercel/pull/9802/files#diff-87f688bc2f474b97f3a8a520f3522827bfe7b7981671ba7f965b78ff045971beR144). I like to name these collections of states for clarity and ease of updating in the future.

I suggest putting this near the top of the file:

```
import type { Build } from '@vercel-internals/types';

// ...

const deploymentProgressStates: Build['readyState'][] = ['QUEUED', 'BUILDING', 'INITIALIZING'];
const deploymentTerminalStates: Build['readyState'][] = ['READY', 'CANCELED', 'ERROR'];
```

and using them in both places, like:

```
if (!deploymentTerminalStates.includes(readyState)) { ... }
```

Reply:
Since the list of states are being referenced in both the `deploy` and `inspect` commands, I thought it made sense to move the state business into a common file. I named it `is-deploying` and it exports a `isDeploying(readyState): boolean` function. I'm definitely open to another solution or different name. What do you think?

Reply:
That's reasonable as well